docs/spec.openapi.yaml

```diff
@@ -0,0 +1,627 @@
+  /register:
+    post:
+      tags:
+        - Authentication
+      description: Register the new user in the service
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/credentials'
+      responses:
+        '200':
+          description: Registration succeed
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/authorization'
+        '409':
+          description: Username already registered
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /challenge:
+    post:
+      tags:
+        - Authorization
+      description: Verifies the received token is still valid
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/authorization'
+      responses:
+        '200':
+          description: Account token is still valid
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/account'
+        '401':
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /account/password:
+    patch:
+      tags:
+        - Account
+      description: Updates the username password
+      requestBody:
+        content:
+          application/json:
+            schema:
+              allOf:
+                - $ref: '#/components/schemas/authorization'
+                - type: object
+                  properties:
+                    currentPassword:
+                      type: string
+                    newPassword:
+                      type: string
+      responses:
+        '200':
+          description: Password updated successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+  /user/uuid/{username}:
+    get:
+      tags:
+        - Account
+      description: Obtain the uuid of an user by its username
```

Comment:
This method isn't required. Usernames are the primary way we identify users. 

This is an internal method offered by the Auth API exclusive for the gateway (java) to use.

See https://github.com/hawks-atlanta/docs/issues/19.